// @tiptap/extension-image/dist/tiptap-extension-image.esm.js
import { Node, mergeAttributes, nodeInputRule } from '@tiptap/core';

const inputRegex = /!\[(.+|:?)]\((\S+)(?:(?:\s+)["'](\S+)["'])?\)/;
const Image = Node.create({
    name: 'image',
    defaultOptions: {
        inline: false,
        HTMLAttributes: {},
    },
    inline() {
        return this.options.inline;
    },
    group() {
        return this.options.inline ? 'inline' : 'block';
    },
    draggable: true,
    addAttributes() {
        return {
            src: {
                default: null,
            },
            alt: {
                default: null,
            },
            title: {
                default: null,
            },
        };
    },
    parseHTML() {
        return [
            {
                tag: 'img[src]',
            },
        ];
    },
    renderHTML({ HTMLAttributes }) {
        return ['img', mergeAttributes(this.options.HTMLAttributes, HTMLAttributes)];
    },
    addCommands() {
        return {
            setImage: options => ({ commands }) => {
                return commands.insertContent({
                    type: this.name,
                    attrs: options,
                });
            },
        };
    },
    addInputRules() {
        return [
            nodeInputRule(inputRegex, this.type, match => {
                const [, alt, src, title] = match;
                return { src, alt, title };
            }),
        ];
    },
});

export default Image;
export { Image, inputRegex };
