// @tiptap/extension-task-list/dist/tiptap-extension-task-list.umd.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@tiptap/core')) :
  typeof define === 'function' && define.amd ? define(['exports', '@tiptap/core'], factory) :
  (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global['@tiptap/extension-task-list'] = {}, global.core));
}(this, (function (exports, core) { 'use strict';

  const TaskList = core.Node.create({
      name: 'taskList',
      defaultOptions: {
          HTMLAttributes: {},
      },
      group: 'block list',
      content: 'taskItem+',
      parseHTML() {
          return [
              {
                  tag: 'ul[data-type="taskList"]',
                  priority: 51,
              },
          ];
      },
      renderHTML({ HTMLAttributes }) {
          return ['ul', core.mergeAttributes(HTMLAttributes, { 'data-type': 'taskList' }), 0];
      },
      addCommands() {
          return {
              toggleTaskList: () => ({ commands }) => {
                  return commands.toggleList('taskList', 'taskItem');
              },
          };
      },
      addKeyboardShortcuts() {
          return {
              'Mod-Shift-9': () => this.editor.commands.toggleTaskList(),
          };
      },
  });

  exports.TaskList = TaskList;
  exports.default = TaskList;

  Object.defineProperty(exports, '__esModule', { value: true });

})));
